### PR TITLE
Implement book label logic

### PR DESCRIPTION
**Pull request by lockieRichter**
Implements the book label logic for the book detail page.

https://github.com/shockbytes/DanteX/assets/26178557/df21a7b0-7c75-47a7-91b7-f15bd20915ff

### CODE REVIEW

**Comment by lockieRichter** [comment]
This doesn't work on web, but that's because the Book Detail page doesn't work on web. Can fix that in the next MR.

File: lib/src/data/book/firebase_book_repository.dart [hunk 1]
```diff
@@ -124,17 +137,25 @@ class FirebaseBookRepository implements BookRepository {
   }
 
   @override
-  Future<void> addLabelToBook(String bookId, BookLabel label) async {
-    final book = await getBook(bookId);
-    book.addLabel(label);
-    return update(book);
+  Future<void> addLabelToBook(String bookId, BookLabel label) {
+    // TODO: implement addLabelToBook
```
Comment (shockbytes) [inline lib/src/data/book/firebase_book_repository.dart]: Why didn't you implement this method?

File: build.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+targets:
```
Comment (shockbytes) [inline build.yaml:1]: Why do we need this file?

File: lib/src/data/book/firebase_book_repository.dart [hunk 3]
```diff
@@ -23,44 +23,50 @@ class FirebaseBookRepository implements BookRepository {
...
+        ).toList();
       },
     );
   }
 
   @override
-  Future<Book> getBook(String id) async {
-    final bookSnapshot = await _rootRef().child(id).get();
-    final bookMap = bookSnapshot.child(id).toMap();
+  Stream<Book> getBook(String id) {
```
Comment (shockbytes) [inline lib/src/data/book/firebase_book_repository.dart:51]: Good call to make this a stream. So we leverage the realtime updates from Firebase.

File: build.yaml (hunk 2 above)
Comment (lockieRichter) [reply build.yaml:1]: This allows `json_serializable` to handle the fact that the `List<BookLabel>` in the `Book` entity is it's own serializeable object.

File: lib/src/data/book/firebase_book_repository.dart (hunk 1 above)
Comment (lockieRichter) [reply lib/src/data/book/firebase_book_repository.dart]: On, I ended up using the updte method, but I'll implement this now anyway and use this instead.

**Comment by lockieRichter** [comment]
All comments addressed

**Review by shockbytes — APPROVED** [review]
Looks good, can't wait to get this merged because then I can finally finish the Realm migration stuff.

**Pull request merged**